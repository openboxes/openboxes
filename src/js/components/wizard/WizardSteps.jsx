import React from 'react';

import _ from 'lodash';
import PropTypes from 'prop-types';

import 'components/wizard/WizardSteps.scss';

const WizardSteps = (props) => (
  <div className="steps-main-box">
    <div className="steps-inside-wrapper">
      {_.map(props.steps, (step, index) => (
        <div
          key={index}
          className={`step-container ${props.currentStep === index + 1 ? 'active' : ''}`}
          data-testid="wizard-step"
          data-stepstate={props.currentStep === index + 1 ? 'active' : 'inactive'}
        >
          <div
            className={props.showStepNumber ? 'circle filled' : 'circle'}
            onClick={() => props.onClick(index + 1)}
            onKeyPress={() => props.onClick(index + 1)}
            role="button"
            tabIndex="0"
            disabled={!props.stepsClickable}
          >
            {props.showStepNumber && <span className="number">{index + 1}</span>}
          </div>
          <div className="step-name">
            {step}
          </div>
        </div>
      ))}
    </div>
  </div>
);

export default WizardSteps;

WizardSteps.propTypes = {
  /** Array of steps names */
  steps: PropTypes.arrayOf(PropTypes.string).isRequired,
  /** Current step number */
  currentStep: PropTypes.number.isRequired,
  /** Function called after clicking on step, (non required) */
  /* eslint-disable-next-line react/no-unused-prop-types */
  onClick: PropTypes.func,
  /** Indicator if steps are clickable (default = false) */
  stepsClickable: PropTypes.bool,
  showStepNumber: PropTypes.bool,
};

WizardSteps.defaultProps = {
  onClick: (stepIdx) => stepIdx,
  stepsClickable: false,
  showStepNumber: false,
};
