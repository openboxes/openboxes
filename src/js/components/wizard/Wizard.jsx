import React, { Component } from 'react';

import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import WizardPage from 'components/wizard/WizardPage';
import WizardSteps from 'components/wizard/WizardSteps';
import WizardTitle from 'components/wizard/WizardTitle';

/** Wizard component. */
class Wizard extends Component {
  constructor(props) {
    super(props);

    this.state = {
      currentPage: this.props.currentPage || 1,
      prevPage: this.props.prevPage || 1,
      values: this.props.initialValues,
    };

    this.nextPage = this.nextPage.bind(this);
    this.prevPage = this.prevPage.bind(this);
    this.goToPage = this.goToPage.bind(this);
  }

  componentWillReceiveProps(nextProps) {
    if (nextProps.currentPage !== this.props.currentPage) {
      this.goToPage(nextProps.currentPage, nextProps.initialValues);
    }
  }

  /**
   * Sets current page state as a previous page and takes user to the next page.
   * @param {object} values
   * @public
   */
  nextPage(values) {
    const { currentPage } = this.state;
    if (this.props.pageList.length > currentPage) {
      this.setState({
        prevPage: currentPage, currentPage: currentPage + 1, values,
      });
      this.props.updateWizardValues(currentPage + 1, values);
    } else {
      this.setState({ values });
      this.props.updateWizardValues(currentPage, values);
    }
  }

  /**
   * Returns user to the previous page.
   * @param {object} values
   * @public
   */
  prevPage(values) {
    const { prevPage } = this.state;
    if (prevPage > 0) {
      this.setState({
        prevPage: prevPage - 1, currentPage: prevPage, values,
      });
      this.props.updateWizardValues(prevPage, values);
    } else {
      this.setState({ values });
      this.props.updateWizardValues(1, values);
    }
  }

  /**
   * Sets current page state as a previous page and takes user to the given number page.
   * @param {object} values
   * @param {number} currentPage
   * @public
   */
  goToPage(currentPage, values) {
    this.setState({ prevPage: currentPage - 1, currentPage, values });
    this.props.updateWizardValues(currentPage, values);
  }

  render() {
    const { currentPage, values } = this.state;
    const {
      title, pageList, stepList, additionalTitle, additionalProps, showStepNumber,
    } = this.props;

    return (
      <div className="content-wrap" data-testid="content-wrap">
        <WizardTitle title={title} additionalTitle={additionalTitle} values={values} />
        <WizardSteps steps={stepList} currentStep={currentPage} showStepNumber={showStepNumber} />
        <div className="panel panel-primary">
          <WizardPage
            pageList={pageList}
            currentPage={currentPage}
            initialValues={values}
            nextPage={this.nextPage}
            prevPage={this.prevPage}
            goToPage={this.goToPage}
            additionalProps={additionalProps}
          />
        </div>
      </div>
    );
  }
}

const mapStateToProps = (state) => ({
  locale: state.session.activeLanguage,
  stockMovementTranslationsFetched: state.session.fetchedTranslations.stockMovement,
  hasPackingSupport: state.session.currentLocation.hasPackingSupport,
});

export default connect(mapStateToProps, {
  // showSpinner, hideSpinner, fetchTranslations,
})(Wizard);

Wizard.propTypes = {
  /** Initial components' data */
  initialValues: PropTypes.shape({
    shipmentStatus: PropTypes.string,
  }),
  title: PropTypes.arrayOf(PropTypes.shape({})).isRequired,
  additionalTitle: PropTypes.oneOf([PropTypes.string, PropTypes.func]),
  currentPage: PropTypes.number.isRequired,
  prevPage: PropTypes.number.isRequired,
  pageList: PropTypes.arrayOf(PropTypes.func).isRequired,
  stepList: PropTypes.arrayOf(PropTypes.string).isRequired,
  updateWizardValues: PropTypes.func,
  additionalProps: PropTypes.shape({}),
  showStepNumber: PropTypes.bool,
};

Wizard.defaultProps = {
  initialValues: {},
  additionalProps: {},
  additionalTitle: null,
  updateWizardValues: () => {},
  showStepNumber: false,
};
