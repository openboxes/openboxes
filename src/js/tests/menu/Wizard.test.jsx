import React from 'react';

import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';

import CreateInvoicePage from 'components/invoice/create/CreateInvoicePage';
import Wizard from 'components/wizard/Wizard';
import WizardPage from 'components/wizard/WizardPage';
import WizardSteps from 'components/wizard/WizardSteps';
import WizardTitle from 'components/wizard/WizardTitle';

import store from '../../store';

let renderedWizard;
let renderedWizardPage;
let renderedWizardSteps;
let renderedWizardTitle;

describe('wizard component', () => {
  beforeEach(() => {
    const props = {
      title: [{ title: '' }],
      currentPage: 1,
      prevPage: 1,
      pageList: [CreateInvoicePage],
      stepList: ['Create', 'Add items', 'Confirm'],
    };
    renderedWizard = renderer.create(
      <Router>
        <Provider
          store={store}
        >
          <Wizard {...props} />
        </Provider>
      </Router>,
    );
  });

  it('should match snapshot', () => {
    expect(renderedWizard.toJSON())
      .toMatchSnapshot();
  });

  it('should render component correctly', () => {
    expect(renderedWizard.root.findByProps({ 'data-testid': 'content-wrap' }))
      .toBeTruthy();
  });
});

describe('wizardPage component', () => {
  beforeEach(() => {
    renderedWizardPage = renderer.create(
      <Router>
        <Provider
          store={store}
        >
          <WizardPage
            pageList={[CreateInvoicePage]}
            nextPage={CreateInvoicePage}
            prevPage={CreateInvoicePage}
            goToPage={CreateInvoicePage}
            currentPage={1}
          />
        </Provider>
      </Router>,
    );
  });

  it('should match snapshot', () => {
    expect(renderedWizardPage.toJSON())
      .toMatchSnapshot();
  });

  it('should render component correctly', () => {
    expect(renderedWizardPage.root.findByProps({ 'data-testid': 'wizardPage' }))
      .toBeTruthy();
  });
});

describe('wizardSteps component', () => {
  beforeEach(() => {
    renderedWizardSteps = renderer.create(
      <Router>
        <Provider
          store={store}
        >
          <WizardSteps steps={['firstTestStep', 'secondTestStep']} currentStep={1} />
        </Provider>
      </Router>,
    );
  });

  it('should match snapshot', () => {
    expect(renderedWizardSteps.toJSON())
      .toMatchSnapshot();
  });

  it('should have an active element', () => {
    expect(renderedWizardSteps.root.findByProps({ 'data-stepstate': 'active' }))
      .toBeTruthy();
  });

  it('should have an inactive element', () => {
    expect(renderedWizardSteps.root.findByProps({ 'data-stepstate': 'inactive' }))
      .toBeTruthy();
  });
});

describe('wizardTitle component', () => {
  beforeEach(() => {
    renderedWizardTitle = renderer.create(
      <Router>
        <Provider store={store}>
          <WizardTitle
            title={[{ title: '' }]}
          />
        </Provider>
      </Router>,
    );
  });

  it('should match snapshot', () => {
    expect(renderedWizardTitle.toJSON())
      .toMatchSnapshot();
  });

  it('should render component correctly', () => {
    expect(renderedWizardTitle.root.findByProps({ 'data-testid': 'wizardTitle' }))
      .toBeTruthy();
  });

  it('should display text correctly', () => {
    render(
      <Router>
        <Provider store={store}>
          <WizardTitle title={[{
            title: '',
            text: 'test',
          }]}
          />
        </Provider>
      </Router>,
    );
    expect(screen.getByText('test'))
      .toBeTruthy();
  });
});
