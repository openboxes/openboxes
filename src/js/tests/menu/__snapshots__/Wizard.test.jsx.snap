// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`wizard component should match snapshot 1`] = `
<div
  className="content-wrap"
  data-testid="content-wrap"
>
  <div
    className="panel-heading movement-number"
    data-testid="wizardTitle"
  >
    <div>
      <span
        style={
          {
            "color": undefined,
          }
        }
      >
        <span
          style={
            {
              "color": "black",
            }
          }
        />
      </span>
    </div>
  </div>
  <div
    className="steps-main-box"
  >
    <div
      className="steps-inside-wrapper"
    >
      <div
        className="step-container active"
        data-stepstate="active"
        data-testid="wizard-step"
      >
        <div
          className="circle"
          disabled={true}
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex="0"
        />
        <div
          className="step-name"
        >
          Create
        </div>
      </div>
      <div
        className="step-container "
        data-stepstate="inactive"
        data-testid="wizard-step"
      >
        <div
          className="circle"
          disabled={true}
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex="0"
        />
        <div
          className="step-name"
        >
          Add items
        </div>
      </div>
      <div
        className="step-container "
        data-stepstate="inactive"
        data-testid="wizard-step"
      >
        <div
          className="circle"
          disabled={true}
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex="0"
        />
        <div
          className="step-name"
        >
          Confirm
        </div>
      </div>
    </div>
  </div>
  <div
    className="panel panel-primary"
  >
    <div
      className="panel-body m-4"
      data-testid="wizardPage"
    >
      <form
        onSubmit={[Function]}
      >
        <div
          className="classic-form with-description"
        >
          <div
            aria-label="Invoice number"
            className="mt-2  mb-0 mx-1 form-group   "
            data-testid="form-field"
          >
            <div
              className="row"
            >
              <label
                className="col-md-2 col-7 col-form-label col-form-label-xs text-center  text-md-right"
                htmlFor="invoiceNumber"
              >
                Invoice number
              </label>
              <div
                className="form-element-container col-md-4 col-7"
              >
                <input
                  className="form-control form-control-xs  "
                  data-testid="input"
                  disabled={true}
                  id="invoiceNumber"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onDragStart={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyPress={[Function]}
                  onKeyUp={[Function]}
                  onWheel={[Function]}
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              aria-label="subtext"
              className="row"
            >
              <div
                className="col-md-2 hidden"
              />
              <div
                className="help-block"
                style={
                  {
                    "float": "left",
                  }
                }
              />
            </div>
          </div>
          <div
            aria-label="Vendor"
            className="mt-2  mb-0 mx-1 form-group required  "
            data-testid="form-field"
          >
            <div
              className="row"
            >
              <label
                className="col-md-2 col-7 col-form-label col-form-label-xs text-center  text-md-right"
                htmlFor="vendor"
              >
                Vendor
              </label>
              <div
                className="form-element-container col-md-4 col-7"
              >
                <div
                  data-testid="custom-select-element"
                  id="select-id_1-container"
                >
                  <div
                    className=""
                    style={
                      {
                        "display": "inline",
                      }
                    }
                    title=" "
                  >
                    <div
                      className="css-1pcexqc-container select-xs undefined"
                      id="select-id_1"
                      onKeyDown={[Function]}
                    >
                      <div
                        className="css-bg1rzq-control react-select__control"
                        onMouseDown={[Function]}
                        onTouchEnd={[Function]}
                      >
                        <div
                          className="css-1hwfws3 react-select__value-container"
                        >
                          <div
                            className="css-151xaom-placeholder react-select__placeholder"
                          />
                          <div
                            className="css-1g6gooi"
                          >
                            <div
                              className="react-select__input"
                              style={
                                {
                                  "display": "inline-block",
                                }
                              }
                            >
                              <input
                                aria-autocomplete="list"
                                autoCapitalize="none"
                                autoComplete="off"
                                autoCorrect="off"
                                disabled={false}
                                id="react-select-2-input"
                                onBlur={[Function]}
                                onChange={[Function]}
                                onFocus={[Function]}
                                spellCheck="false"
                                style={
                                  {
                                    "background": 0,
                                    "border": 0,
                                    "boxSizing": "content-box",
                                    "color": "inherit",
                                    "fontSize": "inherit",
                                    "label": "input",
                                    "opacity": 1,
                                    "outline": 0,
                                    "padding": 0,
                                    "width": "1px",
                                  }
                                }
                                tabIndex="0"
                                type="text"
                                value=""
                              />
                              <div
                                style={
                                  {
                                    "height": 0,
                                    "left": 0,
                                    "overflow": "scroll",
                                    "position": "absolute",
                                    "top": 0,
                                    "visibility": "hidden",
                                    "whiteSpace": "pre",
                                  }
                                }
                              >
                                
                              </div>
                            </div>
                          </div>
                        </div>
                        <div
                          className="css-1wy0on6 react-select__indicators"
                        >
                          <span
                            className="css-bgvzuu-indicatorSeparator react-select__indicator-separator"
                          />
                          <div
                            aria-hidden="true"
                            className="css-16pqwjk-indicatorContainer react-select__indicator react-select__dropdown-indicator"
                            onMouseDown={[Function]}
                            onTouchEnd={[Function]}
                          >
                            <svg
                              aria-hidden="true"
                              className="css-19bqh2r"
                              focusable="false"
                              height={20}
                              viewBox="0 0 20 20"
                              width={20}
                            >
                              <path
                                d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                              />
                            </svg>
                          </div>
                        </div>
                      </div>
                      <input
                        name="select-id_1"
                        type="hidden"
                        value=""
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              aria-label="subtext"
              className="row"
            >
              <div
                className="col-md-2 hidden"
              />
              <div
                className="help-block"
                style={
                  {
                    "float": "left",
                  }
                }
              />
            </div>
          </div>
          <div
            aria-label="Vendor Invoice Number"
            className="mt-2  mb-0 mx-1 form-group required  "
            data-testid="form-field"
          >
            <div
              className="row"
            >
              <label
                className="col-md-2 col-7 col-form-label col-form-label-xs text-center  text-md-right"
                htmlFor="vendorInvoiceNumber"
              >
                Vendor Invoice Number
              </label>
              <div
                className="form-element-container col-md-4 col-7"
              >
                <input
                  className="form-control form-control-xs  "
                  data-testid="input"
                  id="vendorInvoiceNumber"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onDragStart={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyPress={[Function]}
                  onKeyUp={[Function]}
                  onWheel={[Function]}
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              aria-label="subtext"
              className="row"
            >
              <div
                className="col-md-2 hidden"
              />
              <div
                className="help-block"
                style={
                  {
                    "float": "left",
                  }
                }
              />
            </div>
          </div>
          <div
            aria-label="Invoice Date"
            className="mt-2  mb-0 mx-1 form-group required  "
            data-testid="form-field"
          >
            <div
              className="row"
            >
              <label
                className="col-md-2 col-7 col-form-label col-form-label-xs text-center  text-md-right"
                htmlFor="dateInvoiced"
              >
                Invoice Date
              </label>
              <div
                className="form-element-container col-md-4 col-7"
              >
                <div
                  className="date-field"
                >
                  <div
                    className="react-datepicker-wrapper"
                  >
                    <div
                      className="react-datepicker__input-container"
                    >
                      <input
                        autoComplete="off"
                        className="form-control form-control-xs "
                        disabled={false}
                        id="dateInvoiced"
                        onBlur={[Function]}
                        onChange={[Function]}
                        onClick={[Function]}
                        onFocus={[Function]}
                        onKeyDown={[Function]}
                        readOnly={false}
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              aria-label="subtext"
              className="row"
            >
              <div
                className="col-md-2 hidden"
              />
              <div
                className="help-block"
                style={
                  {
                    "float": "left",
                  }
                }
              />
            </div>
          </div>
          <div
            aria-label="Currency"
            className="mt-2  mb-0 mx-1 form-group required  "
            data-testid="form-field"
          >
            <div
              className="row"
            >
              <label
                className="col-md-2 col-7 col-form-label col-form-label-xs text-center  text-md-right"
                htmlFor="currencyUom"
              >
                Currency
              </label>
              <div
                className="form-element-container col-md-4 col-7"
              >
                <div
                  data-testid="custom-select-element"
                  id="select-id_2-container"
                >
                  <div
                    className=""
                    style={
                      {
                        "display": "inline",
                      }
                    }
                    title=" "
                  >
                    <div
                      className="css-1pcexqc-container select-xs undefined"
                      id="select-id_2"
                      onKeyDown={[Function]}
                    >
                      <div
                        className="css-bg1rzq-control react-select__control"
                        onMouseDown={[Function]}
                        onTouchEnd={[Function]}
                      >
                        <div
                          className="css-1hwfws3 react-select__value-container"
                        >
                          <div
                            className="css-151xaom-placeholder react-select__placeholder"
                          />
                          <div
                            className="css-1g6gooi"
                          >
                            <div
                              className="react-select__input"
                              style={
                                {
                                  "display": "inline-block",
                                }
                              }
                            >
                              <input
                                aria-autocomplete="list"
                                autoCapitalize="none"
                                autoComplete="off"
                                autoCorrect="off"
                                disabled={false}
                                id="react-select-3-input"
                                onBlur={[Function]}
                                onChange={[Function]}
                                onFocus={[Function]}
                                spellCheck="false"
                                style={
                                  {
                                    "background": 0,
                                    "border": 0,
                                    "boxSizing": "content-box",
                                    "color": "inherit",
                                    "fontSize": "inherit",
                                    "label": "input",
                                    "opacity": 1,
                                    "outline": 0,
                                    "padding": 0,
                                    "width": "1px",
                                  }
                                }
                                tabIndex="0"
                                type="text"
                                value=""
                              />
                              <div
                                style={
                                  {
                                    "height": 0,
                                    "left": 0,
                                    "overflow": "scroll",
                                    "position": "absolute",
                                    "top": 0,
                                    "visibility": "hidden",
                                    "whiteSpace": "pre",
                                  }
                                }
                              >
                                
                              </div>
                            </div>
                          </div>
                        </div>
                        <div
                          className="css-1wy0on6 react-select__indicators"
                        >
                          <span
                            className="css-bgvzuu-indicatorSeparator react-select__indicator-separator"
                          />
                          <div
                            aria-hidden="true"
                            className="css-16pqwjk-indicatorContainer react-select__indicator react-select__dropdown-indicator"
                            onMouseDown={[Function]}
                            onTouchEnd={[Function]}
                          >
                            <svg
                              aria-hidden="true"
                              className="css-19bqh2r"
                              focusable="false"
                              height={20}
                              viewBox="0 0 20 20"
                              width={20}
                            >
                              <path
                                d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                              />
                            </svg>
                          </div>
                        </div>
                      </div>
                      <input
                        name="select-id_2"
                        type="hidden"
                        value=""
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              aria-label="subtext"
              className="row"
            >
              <div
                className="col-md-2 hidden"
              />
              <div
                className="help-block"
                style={
                  {
                    "float": "left",
                  }
                }
              />
            </div>
          </div>
        </div>
        <div
          className="submit-buttons"
        >
          <button
            className="btn btn-outline-primary float-right btn-xs"
            type="submit"
          >
            Next
          </button>
        </div>
      </form>
    </div>
  </div>
</div>
`;

exports[`wizardPage component should match snapshot 1`] = `
<div
  className="panel-body m-4"
  data-testid="wizardPage"
>
  <form
    onSubmit={[Function]}
  >
    <div
      className="classic-form with-description"
    >
      <div
        aria-label="Invoice number"
        className="mt-2  mb-0 mx-1 form-group   "
        data-testid="form-field"
      >
        <div
          className="row"
        >
          <label
            className="col-md-2 col-7 col-form-label col-form-label-xs text-center  text-md-right"
            htmlFor="invoiceNumber"
          >
            Invoice number
          </label>
          <div
            className="form-element-container col-md-4 col-7"
          >
            <input
              className="form-control form-control-xs  "
              data-testid="input"
              disabled={true}
              id="invoiceNumber"
              onBlur={[Function]}
              onChange={[Function]}
              onDragStart={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyPress={[Function]}
              onKeyUp={[Function]}
              onWheel={[Function]}
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          aria-label="subtext"
          className="row"
        >
          <div
            className="col-md-2 hidden"
          />
          <div
            className="help-block"
            style={
              {
                "float": "left",
              }
            }
          />
        </div>
      </div>
      <div
        aria-label="Vendor"
        className="mt-2  mb-0 mx-1 form-group required  "
        data-testid="form-field"
      >
        <div
          className="row"
        >
          <label
            className="col-md-2 col-7 col-form-label col-form-label-xs text-center  text-md-right"
            htmlFor="vendor"
          >
            Vendor
          </label>
          <div
            className="form-element-container col-md-4 col-7"
          >
            <div
              data-testid="custom-select-element"
              id="select-id_5-container"
            >
              <div
                className=""
                style={
                  {
                    "display": "inline",
                  }
                }
                title=" "
              >
                <div
                  className="css-1pcexqc-container select-xs undefined"
                  id="select-id_5"
                  onKeyDown={[Function]}
                >
                  <div
                    className="css-bg1rzq-control react-select__control"
                    onMouseDown={[Function]}
                    onTouchEnd={[Function]}
                  >
                    <div
                      className="css-1hwfws3 react-select__value-container"
                    >
                      <div
                        className="css-151xaom-placeholder react-select__placeholder"
                      />
                      <div
                        className="css-1g6gooi"
                      >
                        <div
                          className="react-select__input"
                          style={
                            {
                              "display": "inline-block",
                            }
                          }
                        >
                          <input
                            aria-autocomplete="list"
                            autoCapitalize="none"
                            autoComplete="off"
                            autoCorrect="off"
                            disabled={false}
                            id="react-select-6-input"
                            onBlur={[Function]}
                            onChange={[Function]}
                            onFocus={[Function]}
                            spellCheck="false"
                            style={
                              {
                                "background": 0,
                                "border": 0,
                                "boxSizing": "content-box",
                                "color": "inherit",
                                "fontSize": "inherit",
                                "label": "input",
                                "opacity": 1,
                                "outline": 0,
                                "padding": 0,
                                "width": "1px",
                              }
                            }
                            tabIndex="0"
                            type="text"
                            value=""
                          />
                          <div
                            style={
                              {
                                "height": 0,
                                "left": 0,
                                "overflow": "scroll",
                                "position": "absolute",
                                "top": 0,
                                "visibility": "hidden",
                                "whiteSpace": "pre",
                              }
                            }
                          >
                            
                          </div>
                        </div>
                      </div>
                    </div>
                    <div
                      className="css-1wy0on6 react-select__indicators"
                    >
                      <span
                        className="css-bgvzuu-indicatorSeparator react-select__indicator-separator"
                      />
                      <div
                        aria-hidden="true"
                        className="css-16pqwjk-indicatorContainer react-select__indicator react-select__dropdown-indicator"
                        onMouseDown={[Function]}
                        onTouchEnd={[Function]}
                      >
                        <svg
                          aria-hidden="true"
                          className="css-19bqh2r"
                          focusable="false"
                          height={20}
                          viewBox="0 0 20 20"
                          width={20}
                        >
                          <path
                            d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                          />
                        </svg>
                      </div>
                    </div>
                  </div>
                  <input
                    name="select-id_5"
                    type="hidden"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          aria-label="subtext"
          className="row"
        >
          <div
            className="col-md-2 hidden"
          />
          <div
            className="help-block"
            style={
              {
                "float": "left",
              }
            }
          />
        </div>
      </div>
      <div
        aria-label="Vendor Invoice Number"
        className="mt-2  mb-0 mx-1 form-group required  "
        data-testid="form-field"
      >
        <div
          className="row"
        >
          <label
            className="col-md-2 col-7 col-form-label col-form-label-xs text-center  text-md-right"
            htmlFor="vendorInvoiceNumber"
          >
            Vendor Invoice Number
          </label>
          <div
            className="form-element-container col-md-4 col-7"
          >
            <input
              className="form-control form-control-xs  "
              data-testid="input"
              id="vendorInvoiceNumber"
              onBlur={[Function]}
              onChange={[Function]}
              onDragStart={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyPress={[Function]}
              onKeyUp={[Function]}
              onWheel={[Function]}
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          aria-label="subtext"
          className="row"
        >
          <div
            className="col-md-2 hidden"
          />
          <div
            className="help-block"
            style={
              {
                "float": "left",
              }
            }
          />
        </div>
      </div>
      <div
        aria-label="Invoice Date"
        className="mt-2  mb-0 mx-1 form-group required  "
        data-testid="form-field"
      >
        <div
          className="row"
        >
          <label
            className="col-md-2 col-7 col-form-label col-form-label-xs text-center  text-md-right"
            htmlFor="dateInvoiced"
          >
            Invoice Date
          </label>
          <div
            className="form-element-container col-md-4 col-7"
          >
            <div
              className="date-field"
            >
              <div
                className="react-datepicker-wrapper"
              >
                <div
                  className="react-datepicker__input-container"
                >
                  <input
                    autoComplete="off"
                    className="form-control form-control-xs "
                    disabled={false}
                    id="dateInvoiced"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    readOnly={false}
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          aria-label="subtext"
          className="row"
        >
          <div
            className="col-md-2 hidden"
          />
          <div
            className="help-block"
            style={
              {
                "float": "left",
              }
            }
          />
        </div>
      </div>
      <div
        aria-label="Currency"
        className="mt-2  mb-0 mx-1 form-group required  "
        data-testid="form-field"
      >
        <div
          className="row"
        >
          <label
            className="col-md-2 col-7 col-form-label col-form-label-xs text-center  text-md-right"
            htmlFor="currencyUom"
          >
            Currency
          </label>
          <div
            className="form-element-container col-md-4 col-7"
          >
            <div
              data-testid="custom-select-element"
              id="select-id_6-container"
            >
              <div
                className=""
                style={
                  {
                    "display": "inline",
                  }
                }
                title=" "
              >
                <div
                  className="css-1pcexqc-container select-xs undefined"
                  id="select-id_6"
                  onKeyDown={[Function]}
                >
                  <div
                    className="css-bg1rzq-control react-select__control"
                    onMouseDown={[Function]}
                    onTouchEnd={[Function]}
                  >
                    <div
                      className="css-1hwfws3 react-select__value-container"
                    >
                      <div
                        className="css-151xaom-placeholder react-select__placeholder"
                      />
                      <div
                        className="css-1g6gooi"
                      >
                        <div
                          className="react-select__input"
                          style={
                            {
                              "display": "inline-block",
                            }
                          }
                        >
                          <input
                            aria-autocomplete="list"
                            autoCapitalize="none"
                            autoComplete="off"
                            autoCorrect="off"
                            disabled={false}
                            id="react-select-7-input"
                            onBlur={[Function]}
                            onChange={[Function]}
                            onFocus={[Function]}
                            spellCheck="false"
                            style={
                              {
                                "background": 0,
                                "border": 0,
                                "boxSizing": "content-box",
                                "color": "inherit",
                                "fontSize": "inherit",
                                "label": "input",
                                "opacity": 1,
                                "outline": 0,
                                "padding": 0,
                                "width": "1px",
                              }
                            }
                            tabIndex="0"
                            type="text"
                            value=""
                          />
                          <div
                            style={
                              {
                                "height": 0,
                                "left": 0,
                                "overflow": "scroll",
                                "position": "absolute",
                                "top": 0,
                                "visibility": "hidden",
                                "whiteSpace": "pre",
                              }
                            }
                          >
                            
                          </div>
                        </div>
                      </div>
                    </div>
                    <div
                      className="css-1wy0on6 react-select__indicators"
                    >
                      <span
                        className="css-bgvzuu-indicatorSeparator react-select__indicator-separator"
                      />
                      <div
                        aria-hidden="true"
                        className="css-16pqwjk-indicatorContainer react-select__indicator react-select__dropdown-indicator"
                        onMouseDown={[Function]}
                        onTouchEnd={[Function]}
                      >
                        <svg
                          aria-hidden="true"
                          className="css-19bqh2r"
                          focusable="false"
                          height={20}
                          viewBox="0 0 20 20"
                          width={20}
                        >
                          <path
                            d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                          />
                        </svg>
                      </div>
                    </div>
                  </div>
                  <input
                    name="select-id_6"
                    type="hidden"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          aria-label="subtext"
          className="row"
        >
          <div
            className="col-md-2 hidden"
          />
          <div
            className="help-block"
            style={
              {
                "float": "left",
              }
            }
          />
        </div>
      </div>
    </div>
    <div
      className="submit-buttons"
    >
      <button
        className="btn btn-outline-primary float-right btn-xs"
        type="submit"
      >
        Next
      </button>
    </div>
  </form>
</div>
`;

exports[`wizardSteps component should match snapshot 1`] = `
<div
  className="steps-main-box"
>
  <div
    className="steps-inside-wrapper"
  >
    <div
      className="step-container active"
      data-stepstate="active"
      data-testid="wizard-step"
    >
      <div
        className="circle"
        disabled={true}
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        tabIndex="0"
      />
      <div
        className="step-name"
      >
        firstTestStep
      </div>
    </div>
    <div
      className="step-container "
      data-stepstate="inactive"
      data-testid="wizard-step"
    >
      <div
        className="circle"
        disabled={true}
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        tabIndex="0"
      />
      <div
        className="step-name"
      >
        secondTestStep
      </div>
    </div>
  </div>
</div>
`;

exports[`wizardTitle component should match snapshot 1`] = `
<div
  className="panel-heading movement-number"
  data-testid="wizardTitle"
>
  <div>
    <span
      style={
        {
          "color": undefined,
        }
      }
    >
      <span
        style={
          {
            "color": "black",
          }
        }
      />
    </span>
  </div>
</div>
`;
